apps/browser/qml/pages/components/AddToAppGridDialog.qml

```diff
@@ -0,0 +1,16 @@
+import QtQuick 2.0
+import Sailfish.Silica 1.0
+
+BookmarkEditDialog {
+    property string icon
+    property var desktopBookmarkWriter
+    property var view
+
+    //% "Add to App Grid"
+    title: qsTrId("sailfish_browser-he-add_bookmark_to_launcher")
+    canAccept: editedUrl !== "" && editedTitle !== ""
+    onAccepted: {
+        var bookmarkWriter = desktopBookmarkWriter.createObject(view)
```

Comment:
Big fat warning if there is no component passed whilst both cases should be working fine... i.e.
```
if (desktopBookmarkWriter) {
    var bookmarkWriter = desktopBookmarkWriter.createObject(view)
    bookmarkWriter.save(editedUrl, editedTitle, icon)
} else {
    console.log("Cannot save to launcher without bookmark writter!!")
}
```